### ISSUE TITLE
No audio plays at all on iOS when building with v397.2

### ISSUE THREAD

**Issue by dxt-ucl** [Scirra/Construct-bugs#8205]
## Problem description

When building any project using v397.2 , upon exporting to XCode and then running on any iPhone, no audio or sounds play at all. 

## Attach a .c3p

[testAudio.zip](https://github.com/user-attachments/files/16882013/testAudio.zip)

## Steps to reproduce

1. Build the attached project for Xcode using v397.2 and then run it on an iPhone. 

## Observed result

2. Pressing the 'play' button should play a song. This works on project preview, and this works if you build for XCode using v397. However if you build for XCode using c397.2 , it doesn't work, no sound is played. 
3. Not included, but  I have tried a wide range of other sounds and file types and see the same issue. 

## Expected result

Pressing the 'play' button should play a song on iPhone. 

## More details

Works fine for me on Chrome and Microsoft edge, but doesn't work when built on iPhone. 

**First affected release:** 

v397.2

The exact same project still works perfectly in v397 and all prior versions.    

## System details

<details><summary>View details</summary>

Product: Construct 3 r397.2 (stable)
Browser: Edge 128.0.2739.54
Browser engine: Chromium
Context: browser
Operating system: Windows 11
Device type: desktop
Device pixel ratio: 0.800000011920929
Logical CPU cores: 16
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36 Edg/128.0.0.0
Language setting: en-US

Local storage
Storage quota (approx): 558 gb
Storage usage (approx): 2.8 gb (0.5%)
Persistant storage: No

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: no
Supports highp precision: yes
Vendor: Google Inc. (AMD)
Renderer: ANGLE (AMD, AMD Radeon(TM) Graphics (0x00001681) Direct3D11 vs_5_0 ps_5_0, D3D11)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_clip_control
EXT_color_buffer_float
EXT_color_buffer_half_float
EXT_conservative_depth
EXT_depth_clamp
EXT_disjoint_timer_query_webgl2
EXT_float_blend
EXT_polygon_offset_clamp
EXT_render_snorm
EXT_texture_compression_bptc
EXT_texture_compression_rgtc
EXT_texture_filter_anisotropic
EXT_texture_mirror_clamp_to_edge
EXT_texture_norm16
KHR_parallel_shader_compile
NV_shader_noperspective_interpolation
OES_draw_buffers_indexed
OES_sample_variables
OES_shader_multisample_interpolation
OES_texture_float_linear
OVR_multiview2
WEBGL_blend_func_extended
WEBGL_clip_cull_distance
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
WEBGL_multi_draw
WEBGL_polygon_mode
WEBGL_provoking_vertex
WEBGL_stencil_texturing
Audio information
System sample rate: 44100 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

WebM Opus (audio/webm;codecs=opus)
WebM Vorbis (audio/webm;codecs=vorbis)
MPEG-4 Opus (audio/mp4;codecs=opus)
MPEG-4 AAC (audio/mp4;codecs=mp4a.40.2)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav;codecs=1)
Supported encode formats:

WebM Opus (audio/webm;codecs=opus)
MPEG-4 Opus (audio/mp4;codecs=opus)
MPEG-4 AAC (audio/mp4;codecs=mp4a.40.2)
Video information
Supported decode formats:

WebM AV1 (video/webm;codecs=av01.0.00M.08)
WebM VP9 (video/webm;codecs=vp9)
WebM VP8 (video/webm;codecs=vp8)
MPEG-4 AV1 (video/mp4;codecs=av01.0.00M.08)
MPEG-4 H.265 (video/mp4;codecs=hev1.1.2.L93.B0)
MPEG-4 H.264 (video/mp4;codecs=avc1.420034)
Supported encode formats:

WebM AV1 (video/webm;codecs=av01.0.00M.08)
WebM VP9 (video/webm;codecs=vp9)
WebM VP8 (video/webm;codecs=vp8)
WebM H.264 (video/webm;codecs=avc1.420034)
MPEG-4 AV1 (video/mp4;codecs=av01.0.00M.08)
MPEG-4 VP9 (video/mp4;codecs=vp9)
MPEG-4 H.264 (video/mp4;codecs=avc1.420034)

</details>

**Comment by AshleyScirra** [comment]
Which version of iOS is the iPhone running?

**Comment by dxt-ucl** [comment]
@AshleyScirra It's an iPhone SE 2022 running iOS 17.6.1 , I'll dig out some other devices to test on though.

**Comment by dxt-ucl** [comment]
@AshleyScirra Also just tested on an iPhone 13 mini, running iOS 17.5.1, has the same issue

**Comment by dxt-ucl** [comment]
@AshleyScirra Ooft. Yep. That's it. My mistake, sorry for wasting your time. I guess there's some change in 397.2 which makes audio (correctly) not play with silent mode on, vs on previous builds that was somehow being overriden. Thanks for the help!

**Comment by AshleyScirra** [comment]
No worries, glad we figured it out - it is indeed a common confusion. I think for whatever reason, iOS's silent mode only affects some audio playback: audio played via the `<audio>` element, which Construct usually uses for music playback, seems to be allowed to play in silent mode, whereas audio played via the Web Audio API, which Construct usually uses for sound effects, seems to be muted. As we found a bug with using the `<audio>` element on iOS we switched it back to using the Web Audio API, and a side effect of that will be music tracks get muted by silent mode where they weren't muted before. Eventually Safari should fix the bug we worked around and we'll put it back, so it'll change again... unfortunately just a consequence of [Safari generally being a nightmare](https://www.construct.net/en/blogs/ashleys-blog-2/safari-releases-development-1616).
